tests/executors/test_executor_loader.py

```diff
@@ -146,6 +146,11 @@ def test_get_hybrid_executors_from_config(self, executor_config, expected_execut
             assert executors == expected_executors_list
 
     def test_init_executors(self):
+        # We need to init provider config in order to import CeleryExecutor
+        ProvidersManager().initialize_providers_configuration()
+
+        from airflow.providers.celery.executors.celery_executor import CeleryExecutor
+
```

Comment:
Hmm. Strange that should not be needed. We have auto-use fixture to do so:

https://github.com/apache/airflow/blob/main/tests_common/pytest_plugin.py#L1471

```
@pytest.fixture(autouse=True)
def initialize_providers_manager():
    from airflow.providers_manager import ProvidersManager

    ProvidersManager().initialize_providers_configuration()
```

Maybe what is needed here is adding the `cleanup_providers_manager` fixture - just to avoid any side-effecs? 

https://github.com/apache/airflow/blob/320bb38d3f91da1a1a208cc735b91d9a2a533b71/tests_common/pytest_plugin.py#L1490

Reply:
Or maybe there is a scenario that those fixture from `pytest_plugin` have not been loaded somehow and that's why you had problem with it ? 

The fixtures from `tests_common.pytest_plugin` should be loaded via https://github.com/apache/airflow/blob/320bb38d3f91da1a1a208cc735b91d9a2a533b71/tests/conftest.py#L32 -  but maybe there is a case they are not ?

```
pytest_plugins = "tests_common.pytest_plugin"
```

Reply:
AH... Maybe simply you have no `uv sync` or `pip install -e ./providers` installed in your venv - and that's where all the problems are from ?  In such case, providers will not be available in sources of airlfow and Providers manager will not read their configuration.

I guess that might be the problem you had. Seems that those tests heavily depend on whther you are using workspace (or installing both sub-projects in the venv as equivalent of workspace) or not.

Reply:
See : https://github.com/apache/airflow/blob/main/contributing-docs/07_local_virtualenv.rst#creating-virtualenvs-with-uv -> explains both `uv` and `pip` setup for the env.

I guess we should have two variants of the tests and skip one of them depending if the venv has the providers installed or not, or (maybe better) split that tests and only check for presence of built-in providers, and have similar tests in each of the providers - only checking for the presence of the executor coming from the provider. 

In the future when #44511 and when we (likely) separate executors in their own packages, that might be handy (as we will have many more sub-projects - one per each provider).

Reply:
So that would be again side effect of airlfow import running stuff I guess?

Reply:
Looks like caching issue.

Possibly the "cleanup_providers_manager" should (at least temporary) fix the problem? I guess ProvidersManager is initialized in import and then cache is used. 

ProvidersManager uses caching extensively to speed up things, and `cleanup_providers_managers` cleans the cache on top of reloading it:

```python
@pytest.fixture
def cleanup_providers_manager():
    from airflow.providers_manager import ProvidersManager

    ProvidersManager()._cleanup()
    ProvidersManager().initialize_providers_configuration()
    try:
        yield
    finally:
        ProvidersManager()._cleanup()
```

Reply:
I'm pretty sure it's failing before any fixtures are even run - pytest is still importing the test file. If I simply move the body of `initialize_providers_manager` into the root scope of the pytest plugin things are happy.

Reply:
I see. That explains it - yes.

Reply:
That's a key thing here - it's not this specific test, it's the import of CE at the top that blows up. Not sure where I'd even add the cleanup fixture since it's a top level import...

Reply:
Yeah. and that's never going to happen anyway (in reality) because executor is instantiated dynamically not imported. ...